server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java

```diff
@@ -410,7 +410,7 @@ private static ClusterState createBackingIndex(
     public static ClusterState createFailureStoreIndex(
         MetadataCreateIndexService metadataCreateIndexService,
         String cause,
-        Settings settings,
+        Settings nodeSettings,
```

Comment:
I've regretted not naming this `nodeSettings` ever since I added this, so I had this on my TODO for when I happened to touch it. Thanks for changing this :)